internal/workloads/workload_modifier.go

```diff
@@ -200,22 +205,22 @@ func (m *ResourceModifier) ModifyStatefulSet(statefulSet *appsv1.StatefulSet) Mo
 	return m.modifyResource(&statefulSet.Spec.Template, &statefulSet.ObjectMeta)
 }
 
-func (m *ResourceModifier) modifyResource(podTemplateSpec *corev1.PodTemplateSpec, meta *metav1.ObjectMeta) ModificationResult {
-	if hasBeenModified := m.modifyPodSpec(&podTemplateSpec.Spec); !hasBeenModified {
+func (m *ResourceModifier) modifyResource(podTemplateSpec *corev1.PodTemplateSpec, podMeta *metav1.ObjectMeta) ModificationResult {
```

Comment:
issue: this rename is not correct, the second parameter is the meta from the workload type, e.g. if this is called from `ModifyDeployment` it will be the deployment's meta, not the pod meta. Please revert.